launcher/spec/launch_spec.go

```diff
@@ -154,9 +155,22 @@ func (s *LaunchSpec) UnmarshalJSON(b []byte) error {
 		s.SignedImageRepos = append(s.SignedImageRepos, imageRepos...)
 	}
 
-	if val, ok := unmarshaledMap[memoryMonitoringEnable]; ok && val != "" {
-		if boolValue, err := strconv.ParseBool(val); err == nil {
-			s.MemoryMonitoringEnabled = boolValue
+	memVal, memOk := unmarshaledMap[memoryMonitoringEnable]
+	monVal, monOk := unmarshaledMap[monitoringEnable]
+
+	if memOk && monOk {
+		return fmt.Errorf("both %v and %v are specified, only one is permitted", memoryMonitoringEnable, monitoringEnable)
+	} else if memOk && memVal != "" {
```

Comment:
I know we didn't do this before, but why don't we fail if ParseBool errors? So it's just false?

Also, regardless the memVal != "" check is not needed.